core/Requirement/Base_dropdown.php

```diff
@@ -0,0 +1,136 @@
+<?php
+/**
+ * @package     PublishPress\Checklists
+ * @author      PublishPress <help@publishpress.com>
+ * @copyright   copyright (C) 2019 PublishPress. All rights reserved.
+ * @license     GPLv2 or later
+ * @since       1.0.0
+ */
+
+namespace PublishPress\Checklists\Core\Requirement;
+
+defined('ABSPATH') or die('No direct script access allowed.');
+
+class Base_dropdown extends Base_simple implements Interface_required
+{
+    /**
+     * @var string
+     */
+    protected $type = 'dropdown';
+
+
+    /**
+     * Add the requirement to the list to be displayed in the meta box.
+     *
+     * @param array $requirements
+     * @param stdClass $post
+     *
+     * @return array
+     */
+    public function filter_requirements_list($requirements, $post)
+    {
+        // Check if it is a compatible post type. If not, ignore this requirement.
+        if ($post->post_type !== $this->post_type) {
+            return $requirements;
+        }
+
+        $option_name = $this->name;
+        $options     = $this->module->options;
+
+        // The enabled status
+        $enabled = $this->is_enabled();
+
+        // If not enabled, bypass the method
+        if (!$enabled) {
+            return $requirements;
+        }
+
+        // Option names
+        $option_name_dropdown = $this->name . '_dropdown';
+
+        // Get the value
+        $option_value = '';
+        if (isset($this->module->options->{$option_name_dropdown}[$this->post_type])) {
+            $option_value = $this->module->options->{$option_name_dropdown}[$this->post_type];
+        }
+
+
+        // Check value is empty, to skip
+        if (empty($option_name_dropdown)) {
+            return $requirements;
+        }
+
+
+        // The rule
+        $rule = $this->get_option_rule();
+
+        // Register in the requirements list
+        $requirements[ $this->name ] = [
```

Comment:
Please, follow the PSR-12 style here, removing the spaces between the [ ] symbols.